### PR TITLE
Added support for passing a custom `--testMatch` option to Jest

### PR DESCRIPTION
**Pull request by Xiltyn**
### Summary

Resolves: https://github.com/callstack/reassure/issues/314

- Added new option to `yarn reassure measure` and subsequently - `yarn reassure` commands: `--file | -f`
- Added validation for the option by checking whether provided file exists
- Fallback to default if file does not exist

**Important**
Using `TEST_RUNNER_ARGS` will override this feature as it works by adjusting the `defaultArgs` string

### Test plan

1. Clone repo
2. run `yarn`
3. run `yarn reassure -f "examples/native/src/OtherTest.perf-test.tsx"`

Result: Only one test is ran

### Demo

https://user-images.githubusercontent.com/18323849/217242592-9d221c06-944c-45f9-bbcf-accdd63a7589.mov

### CODE REVIEW

**Comment by Xiltyn** [comment]
I would say, that it might be a good idea to wait until we merge the other CLI init PR so that we could improve on the warning log at the very least. It will have to be move to the new `logger` service anyway. 

I'm happy to readjust as it's super minor in terms of changes, after the former PR will have been merged.

**Comment by mdjastrzebski** [comment]
@Xiltyn I think rather than acceping a single file path, we should rather accept a [glob](https://docs.python.org/3/library/glob.html) that would be somehow passed to Jest for evaluation.

**Comment by mdjastrzebski** [comment]
Perhaps, we could use the same `--testMatch` option as Jest.

**Comment by thymikee** [comment]
Yeah, I'd rather start proxying to Jest and when we need to support other test runners some day, we'll rethink this

File: packages/reassure-cli/src/commands/measure.ts
Comment (mdjastrzebski) [inline packages/reassure-cli/src/commands/measure.ts:119]: It seems that Jest does not have an alias for `--testMatch`, if so let be compatible with it.
```suggestion
```

File: packages/reassure-cli/src/commands/measure.ts
Comment (mdjastrzebski) [inline packages/reassure-cli/src/commands/measure.ts:120]: It seems that Jest accepts an array of globs, could we support that as well. Yargs seems to support array type.
https://jestjs.io/docs/cli#--testmatch-glob1--globn

**Comment by mdjastrzebski** [comment]
@Xiltyn this looks simple and elegant. I've left some minor comments connected with matching Jest options slightly more closely.

**Comment by Xiltyn** [comment]
@mdjastrzebski @thymikee I updated the PR based on the suggestions. It's mainly a stylistic change since we've already been using the --testMatch option underneath anyway. It should be clearer now though.

File: packages/reassure-cli/src/commands/measure.ts
Comment (Xiltyn) [reply packages/reassure-cli/src/commands/measure.ts:128]: Yes, but we're currently using a single string so I went with the current pattern and added an improvement issue to actually properly update and optimise the glob that we're using as default. Please see https://github.com/callstack/reassure/issues/363

Comment (Xiltyn) [reply packages/reassure-cli/src/commands/measure.ts:127]: I would like to make a separate PR to adjust that and also introduce the `__perf__` directory as it would reflect the Jest defaults better.

**Comment by thymikee** [comment]
Needs rebase

File: packages/reassure-cli/src/commands/measure.ts
Comment (thymikee) [reply packages/reassure-cli/src/commands/measure.ts:127]: I'm good with separate PR

**Comment by Xiltyn** [comment]
> Needs rebase

Done 👍

**Review by thymikee — APPROVED** [review]

**Pull request merged**
